server/ServerApp.cpp

```diff
@@ -1869,6 +1867,20 @@ int ServerApp::AddPlayerIntoGame(const PlayerConnectionPtr& player_connection, i
                 break;
             }
         }
+        // Assign player to empire if he doesn't have own empire and delegates signle
+        if (delegation.size() == 1 && empire == nullptr) {
+            for (auto e : Empires()) {
+                if (boost::algorithm::to_lower_copy(e.second->PlayerName()) == boost::algorithm::to_lower_copy(delegation.front())) {
```

Comment:
`boost::iequals` might be less verbose?

Reply:
No, I just blindly copied condition from my server code with case insensitive player names.